generic/src/test/java/org/datahem/processor/generic/GenericStreamPipelineTest.java

```diff
@@ -0,0 +1,600 @@
+	public void withSchemaOptionsTest(){
+
+        //BEGIN INPUT
+        JSONObject messageChild = new JSONObject()
+            .put("StringField", "a string")
+            .put("Int32Field", 32) 
+            .put("Int64Field", 64) 
+            .put("DoubleField", 1.1) 
+            .put("FloatField", 1.1) 
+            .put("BoolField", true) 
+            .put("HiddenStringField", "a hidden string");
+
+        JSONObject testPayloadObject = new JSONObject()
+            .put("StringField", "a string")
+            .put("Int32Field", 32) 
+            .put("Int64Field", 64) 
+            .put("DoubleField", 1.1) 
+            .put("FloatField", 1.1) 
+            .put("BoolField", true) 
+            .put("BytesField",  Base64.getEncoder().encodeToString("bytes".getBytes())) 
+            .put("EnumField", 1)
+            .put("messageChild", messageChild)
+            .put("repeatedMessage", new JSONArray()
+                .put(messageChild)
+                .put(messageChild))
+            .put("repeatedString", new JSONArray().put("one").put("two").put("three"))
+            .put("repeatedInt32", new JSONArray().put(32).put(64).put(128))
+            .put("repeatedInt64", new JSONArray().put(64).put(128).put(256))
+            .put("repeatedDouble", new JSONArray().put(1.1).put(1.2).put(1.3))
+            .put("repeatedFloat", new JSONArray().put(1.1).put(1.2).put(1.3))
+            .put("repeatedBool", new JSONArray().put("true").put("false").put("true"))
+            .put("repeatedBytes", new JSONArray()
+                .put(Base64.getEncoder().encodeToString("bytes".getBytes()))
+                .put(Base64.getEncoder().encodeToString("bytes".getBytes()))
+                .put(Base64.getEncoder().encodeToString("bytes".getBytes())))
+            .put("repeatedEnum", new JSONArray().put(0).put(1).put(2))
+            .put("HiddenStringField", "a hidden string")
+            .put("BigQueryTime", "19:00:00")
+            .put("BigQueryDate", "2019-12-03")
+            .put("BigQueryDatetime", "2019-12-03 19:00:00")
+            .put("BigQueryTimestamp", "2019-12-03T21:00:00+02:00")
+            .put("HiddenStringField", "a hidden string");
+            
+
+        String testPayload = testPayloadObject.toString();
+        byte[] payload = testPayload.getBytes(StandardCharsets.UTF_8);
+        PubsubMessage pm = new PubsubMessage(payload, attributes);
+        //END INPUT
+
+        //BEGIN OUTPUT TABLEROW
+        List<TableRow> attributes = new ArrayList<TableRow>();
+        attributes.add(new TableRow().set("key", "source").set("value", "test"));
+        attributes.add(new TableRow().set("key", "uuid").set("value", "123-456-abc"));
+        attributes.add(new TableRow().set("key", "timestamp").set("value", "2013-08-16T23:36:32.444Z"));
+
+        TableRow childMessage = new TableRow()
+                .set("RenamedChildString", "a string")
+                .set("RenamedChildInt32", 32)
+                .set("RenamedChildInt64", 64)
```

Comment:
Why not a real int64 like 79239016081? So that we really make sure it's not limited to 32 bits...

Reply:
agree